tests/conftest.py

```diff
@@ -2060,7 +2060,7 @@ def spark() -> "SparkSession":
         .config("spark.sql.catalog.hive.warehouse", "s3://warehouse/hive/")
         .config("spark.sql.catalog.hive.s3.endpoint", "http://localhost:9000")
         .config("spark.sql.catalog.hive.s3.path-style-access", "true")
-        .config("spark.sql.execution.arrow.pyspark.enabled", "true")
+        .config("spark.sql.execution.arrow.pyspark.enabled", "false")
```

Comment:
this option is needed for other tests

Reply:
@kevinjqliu There is a memory leak error when it's **true**. I was wondering if anyone encountered this before?

Reply:
that's surprising. Can you post the stack trace?

Reply:
Oh, I liked this option so much :D

Reply:
tests/integration/test_inspect_table.py::test_inspect_files[1]
  /home/codespace/.cache/pypoetry/virtualenvs/pyiceberg-FsHa-ZgB-py3.10/lib/python3.10/site-packages/pyspark/sql/pandas/conversion.py:198: UserWarning: toPandas attempted Arrow optimization because 'spark.sql.execution.arrow.pyspark.enabled' is set to true, but has reached the error below and can not continue. Note that 'spark.sql.execution.arrow.pyspark.fallback.enabled' does not have an effect on failures in the middle of computation.
    An error occurred while calling o71.getResult.
  : org.apache.spark.SparkException: Exception thrown in awaitResult: 
        at org.apache.spark.util.SparkThreadUtils$.awaitResult(SparkThreadUtils.scala:56)
        at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:310)
        at org.apache.spark.security.SocketAuthServer.getResult(SocketAuthServer.scala:98)
        at org.apache.spark.security.SocketAuthServer.getResult(SocketAuthServer.scala:94)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:75)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:52)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
        at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
        at java.base/java.lang.Thread.run(Thread.java:1583)
  Caused by: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 1 times, most recent failure: Lost task 0.0 in stage 0.0 (TID 0) (85bf1d94-7064-4319-9250-09ddcbd80af7.internal.cloudapp.net executor driver): java.lang.UnsupportedOperationException: sun.misc.Unsafe or java.nio.DirectByteBuffer.<init>(long, int) not available
        at org.apache.arrow.memory.util.MemoryUtil.directBuffer(MemoryUtil.java:174)
        at org.apache.arrow.memory.ArrowBuf.getDirectBuffer(ArrowBuf.java:229)
        at org.apache.arrow.memory.ArrowBuf.nioBuffer(ArrowBuf.java:224)
        at org.apache.arrow.vector.ipc.WriteChannel.write(WriteChannel.java:133)
        at org.apache.arrow.vector.ipc.message.MessageSerializer.writeBatchBuffers(MessageSerializer.java:303)
        at org.apache.arrow.vector.ipc.message.MessageSerializer.serialize(MessageSerializer.java:276)
        at org.apache.arrow.vector.ipc.message.MessageSerializer.serialize(MessageSerializer.java:237)
        at org.apache.spark.sql.execution.arrow.ArrowConverters$ArrowBatchIterator.$anonfun$next$1(ArrowConverters.scala:119)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
        at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
        at org.apache.spark.sql.execution.arrow.ArrowConverters$ArrowBatchIterator.next(ArrowConverters.scala:122)
        at org.apache.spark.sql.execution.arrow.ArrowConverters$ArrowBatchIterator.next(ArrowConverters.scala:77)
        at scala.collection.Iterator.foreach(Iterator.scala:943)
        at scala.collection.Iterator.foreach$(Iterator.scala:943)
        at

Reply:
@kevinjqliu 
Seems like the issue is related to - https://stackoverflow.com/questions/62109276/errorjava-lang-unsupportedoperationexception-for-pyspark-pandas-udf-documenta/62625252#62625252, thanks @bdilday for pointing this out!

I downgraded to Java 8 in the DockerFile and it didn't work
RUN curl https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-bundle/1.8.11/aws-java-sdk-bundle-1.8.11.jar -Lo /tmp/aws-java-sdk-bundle-1.8.11.jar

Reply:
Another data point, I was able to run this test successfully with my current docker setup, using the default Spark and Docker configuration from the repo.

Reply:
Had an offline discussion with @kevinjqliu , tried using docker system prune --all and then  using .config("spark.sql.execution.arrow.pyspark.enabled", "true")

I still have the "memory leak" issue in my local